pandas/core/resample.py

```diff
@@ -1597,6 +1597,9 @@ def _gotitem(self, key, ndim, subset=None):
             if isinstance(key, list) and self.key not in key and self.key is not None:
                 key.append(self.key)
             groupby = self._groupby[key]
+            # GH 59350: Index is ignored when using the on keyword argument to resample
+            if self.key is not None:
+                groupby.obj = groupby.obj.reset_index(drop=True)
```

Comment:
What happens if L1599 raises an IndexError?

Reply:
I added a test for that case, which passes, although tbh I haven't looked into why it passes :joy: Plmk if that works. If not I'm happy to dig further

Reply:
Thanks for adding the test. I agree that the test is correct and that it hits the case in question, but I am a little mystified as to why the non-sorted index on `groupby.obj` is okay in this case. I will try to look again when I get the chance, but an explanation would be helpful.

Reply:
This is no longer changing in this PR.